src/juju/api.ts

```diff
@@ -344,80 +343,69 @@ export async function fetchAllModelStatuses(
   dispatch: Store["dispatch"],
   getState: () => RootState,
 ) {
-  const queue = new Limiter({ concurrency: 1 });
   let modelErrorCount = 0;
-  modelUUIDList.forEach((modelUUID) => {
-    queue.push(async (done) => {
-      if (isLoggedIn(getState(), wsControllerURL)) {
-        try {
-          const modelWsControllerURL = getModelByUUID(
-            getState(),
+  // Use for/of so that the awaits are blocking so only one model gets polled at a time.
+  for (const modelUUID of modelUUIDList) {
```

Comment:
Most of this is indentation changes except for this for loop and I removed the `done()` calls.